auth/data/src/main/java/com/nibalk/tasky/auth/data/AuthRepositoryImpl.kt

```diff
@@ -1,18 +1,43 @@
 package com.nibalk.tasky.auth.data
 
 import com.nibalk.tasky.auth.data.remote.AuthApi
+import com.nibalk.tasky.auth.data.remote.dto.LoginRequestDto
 import com.nibalk.tasky.auth.data.remote.dto.RegisterRequestDto
 import com.nibalk.tasky.auth.domain.AuthRepository
+import com.nibalk.tasky.auth.domain.model.LoginRequestParams
 import com.nibalk.tasky.auth.domain.model.RegisterRequestParams
+import com.nibalk.tasky.core.data.auth.toAuthInfo
 import com.nibalk.tasky.core.data.networking.safeCall
+import com.nibalk.tasky.core.domain.auth.SessionStorage
 import com.nibalk.tasky.core.domain.util.DataError
 import com.nibalk.tasky.core.domain.util.EmptyResult
 import com.nibalk.tasky.core.domain.util.asEmptyDataResult
+import com.nibalk.tasky.core.domain.util.onSuccess
 
 class AuthRepositoryImpl(
-    private val authApi: AuthApi
+    private val authApi: AuthApi,
+    private val sessionStorage: SessionStorage
 ): AuthRepository {
 
+    override suspend fun login(
+        loginRequestParams: LoginRequestParams
+    ): EmptyResult<DataError.Network> {
+        val response = safeCall {
+            authApi.login(
+                body = LoginRequestDto(
+                    userEmail = loginRequestParams.email,
+                    userPassword = loginRequestParams.password
+                )
+            )
+        }
+        response.onSuccess { result ->
+            if (result != null) {
+                sessionStorage.set(result.toAuthInfo())
+            }
+        }
```

Comment:
In which case could the result be null? If you're using my implementation of the `Result` class, the result can never be null in the success case 🤔